libraries/AP_DDS/Idl/geometry_msgs/msg/PoseStamped.idl

```diff
@@ -2,17 +2,22 @@
 // with input from geometry_msgs/msg/PoseStamped.msg
 // generated code does not contain a copyright notice
 
-#include "Pose.idl"
-#include "Header.idl"
+#include "geometry_msgs/msg/Point.idl"
+#include "geometry_msgs/msg/Quaternion.idl"
+#include "std_msgs/msg/Header.idl"
+struct Pose {
```

Comment:
Do you still need to declare Pose like this?

Reply:
I tried using the upstream IDL files, as specified as a requirement for the issue, but it fails to generate. 
```
Processing the file /home/ryan/Development/ardu_ws/src/ardupilot/libraries/AP_DDS/Idl/geometry_msgs/msg/Pose.idl...
/home/ryan/Development/ardu_ws/src/ardupilot/libraries/AP_DDS/Idl/geometry_msgs/msg/Pose.idl:5:10: fatal error: geometry_msgs/msg/Point.idl: No such file or directory
    5 | #include "geometry_msgs/msg/Point.idl"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ERROR<callPreprocessor>: Preprocessor return an error 1
```

I would expect some waf changes to adjust the include paths. 
Currently, we don't specify the include paths:
```rule=f"{bld.env.MICROXRCEDDSGEN[0]} -cs -replace -d {os.path.join(gen_path, 'generated')} {idl}",```

Can you look into using unmodified IDL files (coped right from your ROS2 `/opt/ros/humble/share/...` directory, and then supplying the necessary include paths to the generator?

Reply:
I've added it. Here is the resultant build directory. 
```
ryan@ryan-sp3:~/Development/ardu_ws/src/ardupilot$ tree build/sitl/libraries/AP_DDS/
build/sitl/libraries/AP_DDS/
├── AP_DDS_Client.cpp.0.d
├── AP_DDS_Client.cpp.0.o
└── generated
    ├── builtin_interfaces
    │   └── msg
    │       ├── Duration.c
    │       ├── Duration.c.0.d
    │       ├── Duration.c.0.o
    │       ├── Duration.h
    │       ├── Time.c
    │       ├── Time.c.0.d
    │       ├── Time.c.0.o
    │       └── Time.h
    ├── geometry_msgs
    │   └── msg
    │       ├── Point.c
    │       ├── Point.c.0.d
    │       ├── Point.c.0.o
    │       ├── Point.h
    │       ├── Pose.c
    │       ├── Pose.c.0.d
    │       ├── Pose.c.0.o
    │       ├── Pose.h
    │       ├── PoseStamped.c
    │       ├── PoseStamped.c.0.d
    │       ├── PoseStamped.c.0.o
    │       ├── PoseStamped.h
    │       ├── Quaternion.c
    │       ├── Quaternion.c.0.d
    │       ├── Quaternion.c.0.o
    │       ├── Quaternion.h
    │       ├── Twist.c
    │       ├── Twist.c.0.d
    │       ├── Twist.c.0.o
    │       ├── Twist.h
    │       ├── TwistStamped.c
    │       ├── TwistStamped.c.0.d
    │       ├── TwistStamped.c.0.o
    │       ├── TwistStamped.h
    │       ├── Vector3.c
    │       ├── Vector3.c.0.d
    │       ├── Vector3.c.0.o
    │       └── Vector3.h
    ├── sensor_msgs
    │   └── msg
    │       ├── BatteryState.c
    │       ├── BatteryState.c.0.d
    │       ├── BatteryState.c.0.o
    │       ├── BatteryState.h
    │       ├── NavSatFix.c
    │       ├── NavSatFix.c.0.d
    │       ├── NavSatFix.c.0.o
    │       ├── NavSatFix.h
    │       ├── NavSatStatus.c
    │       ├── NavSatStatus.c.0.d
    │       ├── NavSatStatus.c.0.o
    │       └── NavSatStatus.h
    └── std_msgs
        └── msg
            ├── Header.c
            ├── Header.c.0.d
            ├── Header.c.0.o
            └── Header.h

9 directories, 54 files
```